src/typescript/sdk/tests/e2e/queries/equality-checks.ts

```diff
@@ -62,6 +63,26 @@ const checkTuples = (args: [string, JsonValue | undefined, JsonValue | undefined
   expect(rows).toEqual(responses);
 };
 
+const checkEventIndex = (
+  row: SwapEventModel | LiquidityEventModel,
+  response: UserTransactionResponse
+) => {
+  const eventsAndIndices = new Map(response.events.map((event, index) => [event.type, index]));
+  // NOTE: This only works because we only submit a single entry function at a time. We expect that
+  // only one type of contract event will exist per transaction. Otherwise, this would fail or
+  // possibly even give false positives/negatives.
```

Comment:
Anyway to get around this?

Pretty likely that aggregators will chain swap calls

Reply:
Sorry, this is specifically just for the SDK tests. I should have clarified that in the comment, I'll just make it clearer. Tbh I might just fix the logic- I can map the events to their types + nonce and that should make it robust.

Reply:
Addressed in 30de1ed63b649bf67f9de880f2ae8256f00650fa, OK to resolve?